src/common/platform/win32/i_main.cpp

```diff
@@ -560,10 +561,12 @@ INT_PTR CALLBACK ErrorPaneProc (HWND hDlg, UINT msg, WPARAM wParam, LPARAM lPara
 			if (LOWORD(wParam) == IDC_BUTTON1) // we pressed the restart button, so run GZDoom again
 			{
 				HMODULE hModule = GetModuleHandleW(NULL);
+				LPWSTR cmdline = (LPWSTR)calloc(wcslen(GetCommandLineW()) + sizeof("-waitforprocess") + 12, sizeof(WCHAR));
 				WCHAR path[MAX_PATH];
-				GetModuleFileNameW(hModule, path, MAX_PATH);
 
-				ShellExecuteW(NULL, L"open", path, GetCommandLineW(), NULL, SW_SHOWNORMAL);
+				swprintf_s(cmdline, wcslen(GetCommandLineW()) + sizeof("-waitforprocess") + 12, L"%s %s %d", GetCommandLineW(), L"-waitforprocess", GetCurrentProcessId());;
```

Comment:
It’s better to calculate the desired buffer size once, and without a magic number, or at least with a comment why it has such value.